### PR TITLE
karn 0.0.2 (new formula)

### PR DESCRIPTION
**Pull request by prydonius**
as requested in prydonius/karn#8

### CODE REVIEW

File: Library/Formula/karn.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class Karn < Formula
+  homepage "https://github.com/prydonius/karn"
+  url "https://github.com/prydonius/karn/archive/v0.0.2.tar.gz"
+  sha256 "eac77b82f7abd9c37e70d01240e8e69893b7497d3dd5ab7b0fb2df04ae1e6529"
+
+  depends_on "go" => :build
+
+  go_resource "github.com/codegangsta/cli" do
+    url "https://github.com/codegangsta/cli",
```
Comment (dunn) [inline Library/Formula/karn.rb]: If you change this to `https://github.com/codegangsta/cli.git` (and similarly for the next one) you won't need to specify `:using => :git`; it'll be able to figure it out.

**Comment by prydonius** [comment]
Thanks for the tip @dunn

**Comment by prydonius** [comment]
Any update on this?

**Comment by chdiza** [comment]
This formula is unlikely to be accepted, because it is self-submitted, for a project that's only six weeks old, and is at version 0.0.2.  Homebrew [explicitly says](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#niche-or-self-submitted-stuff) it doesn't want such things.

**Comment by chdiza** [comment]
Note that I'm not a Homebrew maintainer, and that I don't particularly like the policy that I linked to.

File: Library/Formula/karn.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    mkdir_p buildpath/"src/github.com/prydonius"
+    ln_s buildpath, buildpath/"src/github.com/prydonius/karn"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "cmd/karn/karn.go"
+    bin.install "karn"
+  end
+
+  test do
+    system "#{bin}/karn", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/karn.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
> This formula is unlikely to be accepted, because it is self-submitted, for a project that's only six weeks old, and is at version 0.0.2. Homebrew explicitly says it doesn't want such things.

I'm actually OK with this given it's relatively popular and, being a Go project, this makes it a bit easier to install.

File: Library/Formula/karn.rb (hunk 2 above)
Comment (prydonius) [inline Library/Formula/karn.rb]: Thanks for the suggestion, I've updated the test :)

File: Library/Formula/karn.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  test do
+    File.open(File.join(ENV["HOME"], ".karn.yml"), "w") do |f|
+      f.puts <<-EOS.undent
+        ---
+        #{testpath}:
+          name: Homebrew Test
+          email: test@brew.sh
+      EOS
+    end
+    system "git", "init"
```
Comment (prydonius) [inline Library/Formula/karn.rb]: Is this the correct way of calling the git command inside of the a test environment? It seems to work on my local machine.

File: Library/Formula/karn.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    mkdir_p buildpath/"src/github.com/prydonius"
+    ln_s buildpath, buildpath/"src/github.com/prydonius/karn"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "cmd/karn/karn.go"
+    bin.install "karn"
+  end
+
+  test do
+    File.open(File.join(ENV["HOME"], ".karn.yml"), "w") do |f|
```
Comment (mikemcquaid) [inline Library/Formula/karn.rb]: You can just use `(testpath/".karn.yml").write <<-EOS.undent` here.

File: Library/Formula/karn.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/karn.rb]: Yup.

File: Library/Formula/karn.rb (hunk 4 above)
Comment (prydonius) [inline Library/Formula/karn.rb]: Thanks, didn't realise HOME was set to testpath!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**